src/Appwrite/Auth/OAuth2/Podio.php

```diff
@@ -0,0 +1,192 @@
+                'redirect_uri' => $this->callback
+            ]);
+
+        return $url;
+    }
+
+    /**
+     * @param string $code
+     *
+     * @return array
+     */
+    protected function getTokens(string $code): array
+    {
+        if (empty($this->tokens)) {
+            $this->tokens = \json_decode($this->request(
+                'POST',
+                $this->apiEndpoint . '/oauth/token',
+                ['Content-Type: application/x-www-form-urlencoded'],
+                \http_build_query([
+                    'grant_type' => 'authorization_code',
+                    'code' => $code,
+                    'redirect_uri' => $this->callback,
+                    'client_id' => $this->appID,
+                    'client_secret' => $this->appSecret,
+                    // 'scope' => \implode(' ', $this->getScopes())
+                ])
+            ), true);
+        }
+
+        return $this->tokens;
+    }
+
+    /**
+     * @param string $refreshToken
+     *
+     * @return array
+     */
+    public function refreshTokens(string $refreshToken): array
+    {
+        $this->tokens = \json_decode($this->request(
+            'POST',
+            $this->apiEndpoint . '/oauth/token',
+            ['Content-Type: application/x-www-form-urlencoded'],
+            \http_build_query([
+                'grant_type' => 'refresh_token',
+                'refresh_token' => $refreshToken,
+                'client_id' => $this->appID,
+                'client_secret' => $this->appSecret,
+            ])
+        ), true);
+
+        if (empty($this->tokens['refresh_token'])) {
+            $this->tokens['refresh_token'] = $refreshToken;
+        }
+
+        return $this->tokens;
+    }
+
+    /**
+     * @param string $accessToken
+     *
+     * @return string
+     */
+    public function getUserID(string $accessToken): string
+    {
+        $user = $this->getUser($accessToken);
+
+        return \strval($user['user_id']) ?? '';
+    }
+
+    /**
+     * @param string $accessToken
+     *
+     * @return string
+     */
+    public function getUserEmail(string $accessToken): string
+    {
+        $user = $this->getUser($accessToken);
+
+        return $user['mail'] ?? '';
+    }
+
+    /**
+     * Check if the OAuth email is verified
+     *
+     * @param string $accessToken
+     *
+     * @return bool
+     */
+    public function isEmailVerified(string $accessToken): bool
+    {
+        $user = $this->getUser($accessToken);
+
+        $mails = $user['mails'];
+        $mainMailIndex = \array_search($user['mail'], \array_map(fn($m) => $m['mail'], $mails));
+        $mainMain = $mails[$mainMailIndex];
+
+        if ($mainMain['verified'] ?? false) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param string $accessToken
+     *
+     * @return string
+     */
+    public function getUserName(string $accessToken): string
+    {
+        return '';
```

Comment:
Does the API not provide the name of the user.?

Reply:
Is there another endpoint that can be used? Like a `/profile` endpoint ?

Reply:
Nice idea! Yep, possible. Implemented.

![CleanShot 2022-07-04 at 12 32 36](https://user-images.githubusercontent.com/19310830/177137434-1e501569-c505-4371-9cc7-a94ef6078621.png)